kratos/geometries/tetrahedra_3d_10.h

```diff
@@ -769,6 +770,29 @@ template<class TPointType> class Tetrahedra3D10 : public Geometry<TPointType>
         return rResult;
     }
 
+    /** Tests the intersection of the geometry with
+     * a 3D box defined by rLowPoint and rHighPoint.
+     * The method is only implemented for simple tets
+     * where the faces are planar.
+     *
+     * @param  rLowPoint  Lower point of the box to test the intersection
+     * @param  rHighPoint Higher point of the box to test the intersection
+     * @return            True if the geometry intersects the box, False in any other case.
+     */
+    bool HasIntersection(const Point& rLowPoint, const Point& rHighPoint) const override
+    {
+        if (this->FacesArePlanar()) {
+            return Tetrahedra3D4<TPointType>(
+                this->pGetPoint(0),
+                this->pGetPoint(1),
+                this->pGetPoint(2),
+                this->pGetPoint(3)).HasIntersection(rLowPoint, rHighPoint);
+        } else {
+             KRATOS_ERROR << "\"HasIntersection\" is not implemented for non-planar 10 noded tetrahedra.";
```

Comment:
Maybe instead of an error, we can decompose in 4 tet4 and throw a warning telling the approximation. This will be an intermediate solution, but I am OK with current implementation

Reply:
@pooyan-dadvand suggested a similar approximation by subdivision of the triangular faces into 4 smaller ones. I think the "debate" really is whether we should provide a correct solution for a subset of all tet10s, or an approximation solution that covers all tet10s.

Reply:
I honestly prefer to stick to the current solution, rather than giving an approximation that can be very wrong in many cases.